frontend/UserList.vue

```diff
@@ -0,0 +1,298 @@
+      } else {
+        this.userIndex = -1;
+      }
+
+      this.currentModalName = modalName;
+      this.$modal.show(modalName);
+    },
+    hideModal: function () {
+      if (this.currentModalName) {
+        this.$modal.hide(this.currentModalName);
+      }
+      this.currentModalName = '';
+      this.userIndex = -1;
+      this.currentUser = {};
+    },
+    confirmEditUserModal: function () {
+      // Disable the save button when the user clicks it so they don't
+      // try to save twice. Re-enable it when we get any response back
+      // from the server (even an error).
+      this.disableConfirmButton = true;
+
+      $.ajax({
+        url: "/user/save",
+        method: "POST",
+        contentType: "application/json",
+        data: JSON.stringify(this.currentUser),
+        success: (data) => {
+          this.disableConfirmButton = false;
+
+          var parsed = JSON.parse(data);
+          if (parsed.status === "error") {
+            flashMessage("Error: " + parsed.message, true);
+            return;
+          }
+          // status === "success"
+          console.log('saved: ', parsed);
+          flashMessage(this.currentUser.email + " saved");
+          console.log(this.userIndex);
+          if (this.userIndex === -1) {
+            // We're getting a new user, insert them at the top.
+            this.users = [parsed.user].concat(this.users);
+          } else {
+            // We edited an existing user, replace their row in
+            // `users`.
+            Vue.set(this.users, this.userIndex, parsed.user);
+          }
+
+          this.hideModal();
+        },
+        error: (err) => {
+          this.disableConfirmButton = false;
+
+          console.warn(err.responseText);
+          flashMessage("Server error: " + err.responseText, true);
+        },
+      });
+    },
+    modalOpened: function () {
+      // Add noscroll to body tag so it doesn't scroll while the modal
+      // is shown.
+      $(document.body).addClass('noscroll');
+      this.disableConfirmButton = false;
+
+    },
+    modalClosed: function () {
+      // Allow body to scroll after modal is closed.
+      $(document.body).removeClass('noscroll');
+    },
+    sortByEmail: function () {
+      var order = this.pagingParameters.order;
+      if (order === AscOrder) {
+        order = DescOrder;
+      }
+      else {
+        order = AscOrder;
+      }
+      this.reset();
+      this.pagingParameters.order = order;
+    },
+    setUsers: function (users) {
+      console.log('inside setUsers: ', users);
+      this.users = users;
+    },
+    reset: function () {
+      // reset data properties back to original values
+      this.currentUser = {};
+      this.users = [];
+      this.userIndex = -1;
+      this.disableConfirmButton = false;
+      this.currentModalName = '';
+      this.pagingParameters = {
+        email: "",
+        order: AscOrder,
+        limit: 40
+      }
+    },
+    removeUser: function () {
```

Comment:
Now that I think about it, I don't think we need to expose deleting users through the UI. If someone wants to delete an account, they should just disable that user. WDYT?

Reply:
I was thinking about it this way..

The `disable` flag would most likely be used as a temporary measure. Whereas a "deletion" would probably be something is permanent, and it seems like a more common use case. We could always do a "soft delete", where we set a "deleted" flag.

Reply:
Ah, I had `disabled` in mind as the permanent feature. If we just want to delete folks, which is fine, you should remove the `disabled` flag.

Reply:
Ok. I guess I hadn't understood the purpose of `disabled`. I thought of it as a way to temporarily disable an account (for whatever reasons); I could see some scenarios where that might be useful.

If we just need a way to remove the user, and not what I described above, then just using the `disabled` would work fine. In that case, I can remove the "delete" button from the UI & disable the api endpoint. (while leaving the http handler & model methods for possible future use).